moninag/server/models.py

```diff
@@ -1,30 +1,156 @@
+"""This module holds Server model class"""
 from django.db import models
 
+from registration.models import CustomUser
 
 STATE_CHOICES = (
-    (1, "NotSelected"),
-    (2, "Production"),
-    (3, "Staging"),
+    ('', "NotSelected"),
+    ('prod', "Production"),
+    ('stag', "Staging"),
 )
 
 
 class Server(models.Model):
     """
-    Server Model
-    Fields:
-        id:         Integer - AutoField (primary key)
-        name:       String - Server name
-        address:    String - Server address (ipv4/ipv6/...)
-        state:      String - production state (production/staging/...)
-        user_id:    Integer - user_id for this server
+    Server
+
+    :argument id: int - autogenerated primary key
+    :argument name: str - Server name
+    :argument address: str - Server address
+    :argument state: str - Server state ('NotSelected'/'Production','Staging')
+    :argument user: id - CustomUser id - user owner id
     """
     name = models.CharField(max_length=50)
     address = models.CharField(max_length=200)
-    state = models.CharField(max_length=100, choices=STATE_CHOICES, default=0)
+    state = models.CharField(max_length=20, choices=STATE_CHOICES, default='')
+    user = models.ForeignKey(CustomUser, default=1, on_delete=models.CASCADE)
+
+    @classmethod
+    def create(cls, name, address, state, user_id):
+        """Create and add server to database.
+
+        :param name: str - server name
+        :param address: str - server address
+        :param state: str - server state
+        :param user_id: id - user id
+        :return: Created server for success, None otherwise.
+        """
+
+        try:
+            user = CustomUser.objects.get(id=user_id)
+        except Exception as error:
+            return None
+
+        server = cls(name=name, address=address, state=state, user=user)
+        server.save()
+
+        return server
+
+    @classmethod
+    def update(cls, id, name, address, state, user_id):
+        """Update server data.
+
+        :param id: server id
+        :param name: server name
+        :param address: server address
+        :param state: server state
+        :param user_id: CustomUser id
+        :return: The return value. Updated server for success, None otherwise.
+        """
+
+        # Check if server exist allready
+        if cls.objects.filter(id=id).count() == 0:
```

Comment:
Remove user search unit,
The method receives the existing user_id and server_id